cpp/src/io/csv/csv_gpu.h

```diff
@@ -23,6 +23,112 @@ namespace cudf {
 namespace io {
 namespace csv {
 namespace gpu {
+
+/**
```

Comment:
I think it would help if we had aliases for the row context types (20-bit, 32-bit, packed, other(?)), so it's obvious what kind of context is passed into which function, and what kind is returned. 
This also provides a good spot to specify the content of these contexts.

Reply:
There is really only the "normal" 32-bit version, eg: `row_count*4 + id` (which can be zero extended to 64-bit like any other number) or the 4x20-bit (packed) version. Would a `typedef uint64_t packed_rowcontext_t;` and `typedef uint32_t rowcontext_t` work ?

Reply:
Went with yes: added `rowctx32_t`, `rowctx64_t` and `packed_rowctx_t` . It does make it easier to infer the intent without having to read the comments.